.extlib/simplesamlphp/lib/SimpleSAML/Utils/Config/Metadata.php

```diff
@@ -330,10 +330,10 @@ public static function parseNameIdPolicy($nameIdPolicy)
                 'Format'      => $nameIdPolicy_cf->getString('Format', Constants::NAMEID_TRANSIENT),
                 'AllowCreate' => $nameIdPolicy_cf->getBoolean('AllowCreate', true),
             ];
-            $spNameQualifier = $nameIdPolicy_cf->getString('SPNameQualifier', false);
-            if ($spNameQualifier !== false) {
-                $policy['SPNameQualifier'] = $spNameQualifier;
-            }
+            //$spNameQualifier = $nameIdPolicy_cf->getString('SPNameQualifier', false);
```

Comment:
so reverse engineering this, I think this line:

https://github.com/catalyst/moodle-auth_saml2/blob/MOODLE_39_STABLE/config/authsources.php#L65C5-L65C56

needs to change into something like:

```
'NameIDPolicy' => [
  'Format' => $saml2auth->config->nameidpolicy,
  'AllowCreate' => ???,
  'SPNameQualifier' => $newsettinghere
],
```